src/python/pants/backend/native/tasks/native_compile.py

```diff
@@ -211,18 +224,17 @@ def _compile(self, compile_request):
     NB: This method must arrange the output files so that `collect_cached_objects()` can collect all
     of the results (or vice versa)!
     """
-    sources = compile_request.sources
+    sources, _ = self._split_sources_and_headers(
+      compile_request.sources, compile_request.header_file_extensions)
 
-    if len(sources) == 0 and not any(s for s in sources if not s.endswith(self.HEADER_EXTENSIONS)):
-      # TODO: do we need this log message? Should we still have it for intentionally header-only
-      # libraries (that might be a confusing message to see)?
-      self.context.log.debug("no sources in request {}, skipping".format(compile_request))
```

Comment:
I think informing the user in some way that they just attempted to build something that is empty/ a no-op is useful information to provide at _some_ log level, however, to the best of my knowledge, there is no precedent for logging this type of situation in Pants. Up to you for whether to remove it.

Reply:
The info-level log here is for the purpose of the integration test. I think with @jsirois's [comment](https://github.com/pantsbuild/pants/pull/6802#discussion_r236336113) we can probably make this log a debug log, and then we don't have to decide about it because the user isn't seeing it by default.

Reply:
The above edit was made!